### ISSUE TITLE
Component: InputText

### ISSUE THREAD

**Issue by Exodiel** [primefaces/primereact#4465]
### Describe the bug

I tried to use the prop keyfilter in mobile browser but something is wrong because it does not respect the regex email that I passed in the prop
![Screenshot_2023-05-30-15-19-12-229_com android chrome](https://github.com/primefaces/primereact/assets/32306593/72d7cbc3-8dce-46f6-80f1-356314c18857)
![Screenshot_2023-05-30-15-19-01-683_com android chrome](https://github.com/primefaces/primereact/assets/32306593/61602c6f-2adc-411b-8531-5a4a29b1feb1)

### PrimeReact version

8.7.3

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome Android

**Comment by melloware** [comment]
@Exodiel I don't have any way of testing Android but can you go to the Showcase in your Android and see if it is still an issue in 9.5.0 ?

 Use: https://primereact.org/keyfilter/

**Comment by Exodiel** [comment]
@melloware these are the dependencies with the latest version of primereact
![Screenshot_2023-05-30-15-41-45-317_com android chrome](https://github.com/primefaces/primereact/assets/32306593/d14e0f24-7b9c-47bb-b28b-5be91dea05cc)

**Comment by melloware** [comment]
OK can you now do me a favor in your Android and go here and type `$` or `(` or some character that is being allowed.

https://w3c.github.io/uievents/tools/key-event-viewer.html

For example here is what my `(` looks like in Chrome on Desktop.

![image](https://github.com/primefaces/primereact/assets/4399574/a4e94bd1-7c18-485e-b835-bed0841a9186)

**Comment by melloware** [comment]
I want to see if Android is interpreting these keys somehow differently.

**Comment by Exodiel** [comment]
@melloware apparently they are different
![Screenshot_2023-05-30-15-49-08-034_com android chrome](https://github.com/primefaces/primereact/assets/32306593/1ac92e60-0870-456e-9bf4-911d49108732)

**Comment by melloware** [comment]
ahhh see "Unidentified" that explains why the keyfilter is not working on your Android.  `event.key` .

There are long threads about Androids nightmarish support of the event.key spec you can read about here: https://github.com/substance/substance/issues/982

And here: https://bugs.chromium.org/p/chromium/issues/detail?id=118639

**Comment by Exodiel** [comment]
then the solution would be to perform a manual implementation in the onInput event?

**Comment by melloware** [comment]
Not sure how to fix it other to look for each `event.which` code like `229` as that is all the info you get from Android but `event.which` is deprecated and not recommended for most browser use.
 You would have to find every one of those keys like `(` and use that site i sent you to figure out the `event.which` code for each and filter it out.  Yuck.

You can read more about it here: https://github.com/primefaces/primereact/issues/2101

**Comment by Exodiel** [comment]
@melloware I reply the test and its works perfectlly

**Comment by melloware** [comment]
OK let me submit a PR!

**Comment by francmarin98** [comment]
What would happen with the other components that use keyfilter, for example textarea? At the end of the day it is also an input :)

**Comment by melloware** [comment]
TextArea also got this fix!

**Comment by francmarin98** [comment]
One more question, what happens if the keyboard language changes, was that scenario tested or on IOS?

**Comment by francmarin98** [comment]
The keyboard language, I understand that the distribution of keys and codes change based on the keyboard language.

**Comment by francmarin98** [comment]
code ASCII

**Comment by Exodiel** [comment]
I tried with a different regexp and in some cases does not work, for example, in the regexp does not allow @ but if  I press / caracter and then press @ caracter then allowed
https://stackblitz.com/edit/react-noqgsg-827es3

**Comment by melloware** [comment]
> I tried with a different regexp and in some cases does not work, for example, in the regexp does not allow @ but if I press / caracter and then press @ caracter then allowed https://stackblitz.com/edit/react-noqgsg-827es3

@Exodiel are you referring to on your Android device the `/` then `@` works?   I am trying your example on Chrome Desktop and I can't make the `@` symbol display.

**Comment by melloware** [comment]
OK I would `console.log(event.data)` find out what is in the BeforeInsert Data field?

**Comment by melloware** [comment]
@Exodiel I have a a feeling event.data in that scenario has more than one key so `key.length === 1` is not true and allowing the key...

**Comment by Exodiel** [comment]
![Screenshot_2023-06-02-07-40-10-962_com.android.chrome.jpg](https://github.com/primefaces/primereact/assets/32306593/09eb6733-0592-4ff5-97ca-e01f99449bed)

I tried a differents combinations and always after I pressed / it's allowed type chars dos suppose does nota allowed

**Comment by Exodiel** [comment]
Something interesting is that every char I type is they same code in android
![Screenshot_2023-06-02-07-52-24-887_com.android.chrome.jpg](https://github.com/primefaces/primereact/assets/32306593/759b9252-0cbd-412b-abda-d1915128937d)

![Screenshot_2023-06-02-07-51-57-520_com.android.chrome.jpg](https://github.com/primefaces/primereact/assets/32306593/8762b8d7-f3ab-4394-8559-42a10a547e43)

![Screenshot_2023-06-02-07-52-34-371_com.android.chrome.jpg](https://github.com/primefaces/primereact/assets/32306593/417d9979-da39-4358-af3f-17f7a4fc11f3)

**Comment by melloware** [comment]
Yes for keydown its all 229 but that is why I use BeforeInsert and check the `event.data` field which on Android actually shows you the text pressed.   That is the recommendation from Stack Overflow since Android will never properly implement `onKeyDown`.

**Comment by melloware** [comment]
Not sure I don't have an Android to test with.

**Comment by Exodiel** [comment]
In my case I made this implementation and it works in android
```javascript
if (
  /^[A-Za-z0-9ÁÉÍÓÚáéíóúÑñÄËÏÖÜäëïöüÀÈÌÒÙàèìòùÂÊÎÔÛâêîôû]+$/.test(
    e.target.value
  )
) {
  setValue(e.target.value);
} else {
  setValue(value.length === 1 ? '' : value);
}
```
